### PR TITLE
AL/MIN-387. Fix RPC liquidityPoolState

### PR DESCRIPTION
**Pull request by lut1k**
**Description:**

Fixed RPC `liquidityPoolState`.
Now this method returns current data.

**Changes in storage or API:**

--

**Checklist**

- [x] Code is covered with tests
- [x] Code is covered with benchmarks
- [x] Fresh benchmark results included

### CODE REVIEW

**Review by ales-by — APPROVED** [review]

File: runtime/src/tests/mod.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (Apollo50) [inline runtime/src/tests/mod.rs:1]: You could remove "tests" directory.

File: pallets/controller/src/lib.rs [hunk 2]
```diff
@@ -468,110 +468,99 @@ pub mod module {
...
+			.filter(|&underlying_id| T::LiquidityPoolsManager::pool_exists(underlying_id))
+			.try_fold(
+				(Balance::zero(), Balance::zero()),
+				|(mut acc_user_supply_in_usd, mut acc_user_borrowed_in_usd),
+				 &pool_id|
+				 -> result::Result<(Balance, Balance), DispatchError> {
+					let wrapped_id = pool_id.wrapped_asset().ok_or(Error::<T>::PoolNotFound)?;
+
+					// Check if user has / had borrowed wrapped tokens in the pool
+					let user_wrap_balance = T::MultiCurrency::free_balance(wrapped_id, &who);
```
[the remark this answers is not in the document]
Comment (lut1k) [reply pallets/controller/src/lib.rs:527]: Fixed

File: pallets/controller/src/lib.rs [hunk 3]
```diff
@@ -468,110 +468,99 @@ pub mod module {
...
+							pool_data.total_borrowed,
+						)?;
+						acc_user_supply_in_usd += Rate::from_inner(user_wrap_balance)
+							.checked_mul(&current_exchange_rate)
+							.and_then(|v| v.checked_mul(&oracle_price))
+							.map(|x| x.into_inner())
+							.ok_or(Error::<T>::BalanceOverflow)?;
+					}
+					if has_user_borrow_balance {
+						let user_borrow_balance =
```
[the remark this answers is not in the document]
Comment (lut1k) [reply pallets/controller/src/lib.rs:553]: Fixed to `has_user_borrow_underlying_balance`

File: pallets/controller/src/lib.rs [hunk 4]
```diff
@@ -580,70 +569,52 @@ impl<T: Config> Pallet<T> {
...
+				let current_exchange_rate = <LiquidityPools<T>>::get_exchange_rate_by_interest_params(
+					pool_id,
 					pool_data.total_protocol_interest,
 					pool_data.total_borrowed,
 				)?;
+				let wrapped_id = pool_id.wrapped_asset().ok_or(Error::<T>::NotValidUnderlyingAssetId)?;
+				let pool_supply_wrapped_balance = T::MultiCurrency::total_issuance(wrapped_id);
 				let oracle_price = T::PriceSource::get_underlying_price(pool_id).ok_or(Error::<T>::InvalidFeedPrice)?;
-				let pool_total_value_usd = Rate::from_inner(wrapped_balance)
+				let pool_supply_underlying_balance_usd = Rate::from_inner(pool_supply_wrapped_balance)
```
[the remark this answers is not in the document]
Comment (lut1k) [reply pallets/controller/src/lib.rs:587]: Fixed to `pool_supply_balance_in_usd`

File: pallets/controller/src/lib.rs [hunk 5]
```diff
@@ -723,44 +693,96 @@ impl<T: Config> Pallet<T> {
 	/// - `who`: The address whose balance should be calculated.
 	/// - `underlying_asset`: ID of the currency, the balance of borrowing of which we calculate.
 	/// - `pool_borrow_index`: borrow index for the pool
-	fn calculate_borrow_balance(
+	fn calculate_user_borrow_balance(
```
[the remark this answers is not in the document]
Comment (lut1k) [reply pallets/controller/src/lib.rs:696]: Added

File: runtime/src/tests/mod.rs (hunk 1 above)
Comment (lut1k) [reply runtime/src/tests/mod.rs:1]: Removed

**Review by Jekshmek — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]

**Pull request merged**